### PR TITLE
feat: Use jpype instead of subprocess

### LINKED ISSUE #352: Use JPype instead of subprocess
**Is your feature request related to a problem? Please describe.**
Repeatedly calling the Jar as java subprocess causes overhead (start java, load modules, etc.)

**Describe the solution you'd like**
Switch to JPype to launch the VM only once and call into the APIs directly.
For a start, the CLI main functions could be used to simplify the transition, maybe switching to the actual API level eventually.

**Describe alternatives you've considered**
Sticking with subprocess is possible but has the drawbacks described above.

**Additional context**
The python-pdfbox project had a similar transition from subprocess to JPype.

### PR DESCRIPTION
**Pull request by chezou**
## Description
Close #352 
Successor of #355 

## Motivation and Context
To reduce JVM launch overhead for each execution.

## How Has This Been Tested?
Unit test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I read [the contributing document](https://tabula-py.readthedocs.io/en/latest/contributing.html).
- [x] My code follows the code style of this project with running linter.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -22,4 +22,6 @@ def lint(session):
 @nox.session
 def tests(session):
     session.install(".[test]")
-    session.run("pytest", "-v")
+    session.run("pytest", "-v", "tests/test_read_pdf_table.py")
+    session.run("pytest", "-v", "tests/test_read_pdf_jar_path.py")
```
Comment (chezou) [inline noxfile.py:26]: This is due to jpype limitation that not allow shutdown JVM. Hence, we need to make separate session run for JVM option testing.

**Comment by chezou** [comment]
The comparison of test running time is roughly 3 times faster than subprocess.

master branch:

```
$ Measure-Command { pytest .\tests\test_read_pdf_table.py }

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 32
Milliseconds      : 298
Ticks             : 322984726
TotalDays         : 0.000373824914351852
TotalHours        : 0.00897179794444444
TotalMinutes      : 0.538307876666667
TotalSeconds      : 32.2984726
TotalMilliseconds : 32298.4726
```

This PR:

```
 Measure-Command { pytest .\tests\test_read_pdf_table.py }

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 9
Milliseconds      : 860
Ticks             : 98607644
TotalDays         : 0.000114129217592593
TotalHours        : 0.00273910122222222
TotalMinutes      : 0.164346073333333
TotalSeconds      : 9.8607644
TotalMilliseconds : 9860.7644
```

File: noxfile.py (hunk 1 above)
Comment (mara004) [reply noxfile.py:26]: > This is due to jpype limitation that doesn not allow shutdown JVM

how about `jpype.shutdownJVM()`?

Comment (chezou) [reply noxfile.py:26]: According to the document, we have to reboot Python to relaunch JVM.

https://jpype.readthedocs.io/en/latest/api.html#jpype.shutdownJVM
> Due to limitations in the JPype, it is not possible to restart the JVM after being terminated.

Comment (mara004) [reply noxfile.py:26]: Ah, I see.

**Pull request merged**